src/runner.jl

```diff
@@ -375,60 +255,26 @@ function test!(pkg::AbstractString,
     # Copied from Pkg.test approach
     julia_args = Cmd(julia_args)
     test_args = Cmd(test_args)
-    pkgspec = deepcopy(PackageSpec(pkg))
-    ctx = Context()
-    
-    if !isinstalled!(ctx, pkgspec)
-        push!(nopkgs, pkgspec.name)
-        return
-    end
-    Pkg.instantiate(ctx)
-    testfilepath = gettestfilepath(ctx, pkgspec)
-
+    ctx, pkgspec = try
+        TestEnv.ctx_and_pkgspec(pkg)
```

Comment:
ideally we wouldn't use the TestEnv internals here.
Because we would like to contain all the evil of accessing interals of Pkg into TestEnv so TestReports can be safe from it

Reply:
but i don't think should block on this

Reply:
Added a TODO and will make an issue after merge